### PR TITLE
case sensitive icon in filter

### LINKED ISSUE #426: Have the possibility for filtering in Taipy table to be case insensitive
**Description**
This will add the possibility to choose whether the filter is case-sensitive or not in Taipy tables (when the filter is on).

**Acceptance Criteria**
- [ ] Ensure new code is unit tested, and check code coverage is at least 90%
- [ ] Propagate any change on the demos and run all of them to ensure there is no breaking change
- [ ] Ensure any change is well documented

### LINKED ISSUE #2215: [🐛 BUG] Filtering on string in tables has a bad icon
### What went wrong? 🤔

Since [PR#2087](https://github.com/Avaiga/taipy/pull/2087), addressing #426, there's an icon in the string field indicating whether or not the filtering should take into account the casing.

This icon is ugly:
image

### Expected Behavior

A more explicit icon (which should not be difficult to find) is visible.

### Version of Taipy

' develop' branch at the time of creating this issue.

### Acceptance Criteria

- [ ] A unit test reproducing the bug is added.
- [ ] Any new code is covered by a unit tested.
- [ ] Check code coverage is at least 90%.
- [ ] The bug reporter validated the fix.
- [ ] Related issue(s) in taipy-doc are created for documentation and Release Notes are updated.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by FredLL-Avaiga**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
Replace case sensitive switch with icon by a simple icon changing colors

## Related Tickets & Documents

resolves #2215
resolves #426

## How to reproduce the issue

display a table with filter or a datanode_selector or scenario_selector

## Checklist
- [x] Does this solution meet the acceptance criteria of the related issue?
- [ ] Is the related issue checklist completed?
- [ ] Does this PR adds unit tests for the developed code? If not, why?
- [ ] End-to-End tests have been added or updated?
- [x] Was the documentation updated, or a dedicated issue for documentation created? (If applicable)
- [ ] Is the release notes updated? (If applicable)

### CODE REVIEW

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 1]
```diff
@@ -305,14 +304,11 @@ const FilterRow = (props: FilterRowProps) => {
...
                                 endAdornment: (
-                                    <Switch
-                                        onChange={toggleMatchCase}
-                                        checked={matchCase}
-                                        size="small"
-                                        checkedIcon={<MatchCase />}
-                                        icon={<MatchCase color="disabled" />}
-                                        inputProps={{ "aria-label": "Case Sensitive Toggle" }}
-                                    />
+                                    <Tooltip title={matchCase ? "case sensitive" : "case insensitive"}>
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]: Are tooltip text usually NOT capitalized?

**Review by FabienLelaquais — DISMISSED** [review]
👍

File: frontend/taipy-gui/src/components/Taipy/TableFilter.tsx [hunk 2]
```diff
@@ -305,14 +304,11 @@ const FilterRow = (props: FilterRowProps) => {
...
                                 endAdornment: (
-                                    <Switch
-                                        onChange={toggleMatchCase}
-                                        checked={matchCase}
-                                        size="small"
-                                        checkedIcon={<MatchCase />}
-                                        icon={<MatchCase color="disabled" />}
-                                        inputProps={{ "aria-label": "Case Sensitive Toggle" }}
-                                    />
+                                    <Tooltip title={matchCase ? "Case Sensitive" : "Case Insensitive"}>
```
Comment (FabienLelaquais) [inline frontend/taipy-gui/src/components/Taipy/TableFilter.tsx]:
```suggestion
                                    <Tooltip title={matchCase ? "Case sensitive" : "Case insensitive"}>
```

File: tests/gui_core/test_context_filter.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        assert len(res) == 1
+        assert res[0] is scenario_a
+
+        res = gui_core_context.get_filtered_scenario_list(
+            scenarios, [{"col": "config_id", "type": "str", "value": "_A_", "action": "contains", "matchCase": True}]
+        )
+        assert len(res) == 0
+
+
+datanode_a = PickleDataNode("datanode_a_config_id", Scope.SCENARIO)
```
Comment (FabienLelaquais) [inline tests/gui_core/test_context_filter.py:58]: Wouldn't tests run faster with in-memory data nodes?

**Review by FabienLelaquais — DISMISSED** [review]
I know you're going to pretend I'm meticulous or even obsessive...

File: tests/gui_core/test_context_filter.py (hunk 3 above)
Comment (FredLL-Avaiga) [reply tests/gui_core/test_context_filter.py:58]: we don't write/read it

Comment (FredLL-Avaiga) [reply tests/gui_core/test_context_filter.py:58]: but if you have a preferred type ...

**Review by FabienLelaquais — APPROVED** [review]
👍

**Pull request merged**